src/goose/brokers/redis/batch.clj

```diff
@@ -70,3 +58,72 @@
        :retrying    retrying
        :successful  successful
        :dead        dead})))
+
+(defn set-batch-expiration
+  [conn id linger-in-hours]
```

Comment:
`linger-in-hours` is optional, so this can be a multi-arity function, & when called with 2 args, use the default value.

Reply:
The callers would pass a nil value, I don't think we should make this multi-arity.

Also, the value of linger read from Redis would be a string, so changed this arg to `linger-sec-str.`
And both callers of the fn, `post-batch-exec` and the fn we would implement for delete API would get the values from Redis. And I don't thing the callers should be expected to do a nil check and parse to long. This can be done by `set-batch-expiration` itself.